src/tree.rs

```diff
@@ -107,11 +107,19 @@ impl MerklePath {
 
     /// Instantiates a new Merkle path given a leaf position and authentication path.
     pub(crate) fn new(position: u32, auth_path: [pallas::Base; MERKLE_DEPTH_ORCHARD]) -> Self {
-        Self {
+        Self::from_parts(
             position,
-            auth_path: gen_const_array_with_default(MerkleHashOrchard::empty_leaf(), |i| {
+            gen_const_array_with_default(MerkleHashOrchard::empty_leaf(), |i| {
                 MerkleHashOrchard(auth_path[i])
             }),
+        )
+    }
+
+    /// Instantiates a new Merkle path given a leaf position and authentication path.
+    pub fn from_parts(position: u32, auth_path: [MerkleHashOrchard; MERKLE_DEPTH_ORCHARD]) -> Self {
```

Comment:
```markdown
### Added
- `MerklePath::from_parts`
```
This is a slightly awkward API (like all "from parts" APIs) in that the caller could construct this with an invalid `(position, auth_path)` tuple. But the effects of that is relatively limited: if the resulting path is used in `Builder` (which is the only in-crate consumer of `MerklePath`), it will result in a bundle with an invalid proof.

Reply:
Yeah. Constructing from parts is the lowest-common-denominator and relies only on builtin types; any additional structure we might lay over this would more or less just obfuscate the fundamental operation you're having to implement anyway.